include/deal.II/fe/mapping_fe.h

```diff
@@ -0,0 +1,741 @@
+
+#ifndef __deal2__mapping_fe_h
+#define __deal2__mapping_fe_h
+
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/table.h>
+#include <deal.II/fe/mapping.h>
+#include <deal.II/fe/fe.h>
+#include <deal.II/base/qprojector.h>
+#include <deal.II/base/thread_management.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+/*!@addtogroup mapping */
+/*@{*/
+
+/**
+ * The MappingFE is a generalization of the MappingQEulerian class, for arbitrary
+ * vectorial finite elements. The main difference is that this class uses a vector
+ * of absolute positions instead of a vector of displacement.
+ * In particular we think of a collections of a FE_Q or
+ * Bezier finite element (FE_Bernstein) repeated a number of times equal to the space
+ * dimension. The idea is to construct the mapping using a vector of control
+ * points, a DoFHandler associated to the geometry of the problem and a
+ * ComponentMask that tells us which components to use for the mapping.
+ * This mapping will grab from the DoFHandler the finite element, or better
+ * the collection of finite elements, to compute the mapping shape functions.
+ * So we will have two different Finite Element and DoFHandler, one for the
+ * solution field and one to describe the geometry of the problem. Historically
+ * in the deal.II library there was not such a distinction. The differences
+ * between this mapping and the MappingQ class are quite important.
+ * The MappingFE, being a generalization, requires a higher level of abstraction.
+ * This is the reason why it takes a DoFHandler and a vector of control points
+ * that are the coefficients of the shape function (so in general it is a vector
+ * of coefficient).
+ *
+ *
+ * Typically, the DoFHandler operates on a finite element that is constructed
+ * as a system element (FESystem) from continuous FE_Q() objects. An example
+ * is shown below:
+ * @code
+ *    const FE_Q<dim,spacedim> feq(1);
+ *    const FESystem<dim,spacedim> fesystem(feq, spacedim);
+ *    DoFHandler<dim,spacedim> dhq(triangulation);
+ *    dhq.distribute_dofs(fesystem);
+ *    Vector<double> eulerq(dhq.n_dofs());
+ *    const ComponentMask mask(spacedim, true);
+ *    MappingFE<dim,spacedim> map(eulerq, dhq, mask);
+ *    map.update_euler_vector_using_triangulation(eulerq);
+ * @endcode
+
+
+ *
+ * @author Luca Heltai, Marco Tezzele 2013, 2015
+ */
+template <int dim, int spacedim=dim,
+          class DH=DoFHandler<dim,spacedim>,
+          class VECTOR=Vector<double> >
+class MappingFE : public Mapping<dim,spacedim>
+{
+public:
+  /**
+   * Constructor. The first argument is a VECTOR that specifies the
+   * transformation of the domain from the reference to the current
+   * configuration. This is filled calling the method
+   * update_euler_vector_using_triangulation.
+   */
+  MappingFE (const VECTOR  &euler_vector,
+             const DH      &euler_dof_handler,
+             const ComponentMask mask=ComponentMask());
```

Comment:
I don't understand the last sentence of the documentation. You are passing a vector here. What happens to this vector and why do I need to call another function with it?

By the way, the vector is associated with the DoFHandler. I think it would be appropriate to reorder these two.